Plots/Polygons/NCL_polyg_2.py

```diff
@@ -0,0 +1,138 @@
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/polyg_2_lg.png
+"""
+
+###############################################################################
+# Import packages:
+# ----------------
+import xarray as xr
+import cartopy.crs as ccrs
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+import matplotlib.cm as cm
+from geocat.viz import util as gvutil
+
+import geocat.datafiles as gdf
+import cartopy.io.shapereader as shpreader
+import shapely.geometry as sgeom
+from mpl_toolkits.axes_grid1.inset_locator import inset_axes
+
+###############################################################################
+# Initialize color map and bounds for each color
+
+colormap = colors.ListedColormap(['mediumpurple', 'mediumblue', 'royalblue', 'cornflowerblue', 'lightblue', 'lightseagreen', 'yellowgreen',
+                                  'green', 'wheat', 'tan', 'gold', 'orange', 'red', 'firebrick'])
+
+# Values represent average number of inches of rain
+colorbounds = [0, 5, 10, 15, 20, 25, 30, 35, 40, 50, 60, 70, 80, 90, 100]
+
+###############################################################################
+# Define helper function to determine which color to fill the divisions based on precipitation data
+
+
+def findDivColor(colorbounds, pdata):
+
+    for x in range(len(colorbounds)):
+
+        if pdata >= colorbounds[len(colorbounds)-1]:
+            return colormap.colors[x-1]
+        if pdata >= colorbounds[x]:
+            continue
+        else:
+            # Index is 'x-1' because colorbounds is one item longer than colormap
+            return colormap.colors[x-1]
+
+
+###############################################################################
+# Plot map
+
+# Create plot figure
+fig = plt.figure(figsize=(10, 10))
+
+# Add a subplot for lambert conformal map
+ax = fig.add_subplot(111, projection=ccrs.LambertConformal(), frameon=False, xbound=0.0, ybound=0.0)
+
+# Set latitude and longitude extent of map
+ax.set_extent([-119, -74, 18, 50], ccrs.Geodetic())
+
+# Set shape name of map (which depicts the United States)
+shapename = 'admin_1_states_provinces_lakes_shp'
+states_shp = shpreader.natural_earth(resolution='110m', category='cultural', name=shapename)
+
+# Set title and title fontsize of plot using gvutil function instead of matplotlib function call
+gvutil.set_titles_and_labels(ax, maintitle="\nAverage Annual Precipiation \n Computed for the period 1899-1999 \n NCDC climate division data \n \n",
+                             maintitlefontsize=18)
+
+# Add outlines of each state within the United States
+for state in shpreader.Reader(states_shp).geometries():
+
+    facecolor = 'white'
+    edgecolor = 'black'
+
+    ax.add_geometries([state], ccrs.PlateCarree(), facecolor=facecolor, edgecolor=edgecolor)
+
+# Open climate division datafile and add to xarray
+ds = xr.open_dataset(gdf.get("netcdf_files/climdiv_prcp_1899-1999.nc"), decode_times=False)
```

Comment:
How about moving this data reading block to a place just after import packages block so that we can have a code flow like:

import packages -> read data -> helper functions -> plot